Mathlib/CategoryTheory/Limits/VanKampen.lean

```diff
@@ -0,0 +1,501 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.CommSq
+import Mathlib.CategoryTheory.Limits.Shapes.StrictInitial
+import Mathlib.CategoryTheory.Limits.Shapes.Types
+import Mathlib.Topology.Category.TopCat.Limits.Pullbacks
+import Mathlib.CategoryTheory.Limits.FunctorCategory
+import Mathlib.CategoryTheory.Limits.Constructions.FiniteProductsOfBinaryProducts
+
+#align_import category_theory.extensive from "leanprover-community/mathlib"@"178a32653e369dce2da68dc6b2694e385d484ef1"
+
+/-!
+
+# Universal colimits and van Kampen colimits
+
+## Main definitions
+- `CategoryTheory.IsVanKampenColimit`: A (colimit) cocone over a diagram `F : J ⥤ C` is universal
+  if it is stabler under pullbacks.
+- `CategoryTheory.IsUniversalColimit`: A (colimit) cocone over a diagram `F : J ⥤ C` is van
+  Kampen if for every cocone `c'` over the pullback of the diagram `F' : J ⥤ C'`,
+  `c'` is colimiting iff `c'` is the pullback of `c`.
```

Comment:
* The lean names and the docstrings seem to be swapped
* `stabler` should be `stable`
* Could you please add a reference to literature on Van Kampen colimits?

Reply:
Done.